internal/engine/upper.go

```diff
@@ -853,20 +853,23 @@ func handleServiceEvent(ctx context.Context, state *store.EngineState, action Se
 	ms.LBs[k8s.ServiceName(service.Name)] = action.URL
 }
 
-func handleK8SEvent(ctx context.Context, state *store.EngineState, event *v1.Event) {
-	v, ok := state.ObjectsByK8SUIDs[k8s.UID(event.InvolvedObject.UID)]
+func handleK8SEvent(ctx context.Context, state *store.EngineState, action store.K8SEventAction) {
+	evt := action.Event
+	v, ok := state.ObjectsByK8SUIDs[k8s.UID(evt.InvolvedObject.UID)]
 	if !ok {
 		return
 	}
 
-	if event.Type == "Warning" {
-		// TODO: log this to the manifest's log as well
-		logger.Get(ctx).Infof("event received for %s: %s:%s:%s: %s",
-			v.Manifest,
-			v.Entity.Name(),
-			v.Entity.Namespace(),
-			v.Entity.Kind,
-			event.Message)
+	if evt.Type != "Normal" {
+		ms, ok := state.ManifestState(v.Manifest)
+		if !ok {
+			return
+		}
+
+		ms.CombinedLog = model.AppendLog(ms.CombinedLog, action, state.LogTimestamps)
+		logger.Get(ctx).Infof("%s%s", logPrefix(ms.Name.String()), action.MessageRaw())
```

Comment:
Is this supposed to be here?

Reply:
Yeahhhh it's how we get it into the global log :( unless you know a better way?